### PR TITLE
Software Gardening Part 2 — v0.5.1 maintenance release

### PR DESCRIPTION
**Pull request by gwaybio**
- **Removed dead code.** We deleted the crop_generators/ plugin dispatch system (replaced by a direct import of SingleImageCropGenerator), the dataset/sampling.py single-cell export module, training-related methods from CropGenerator, and the comet_ml/dead imports from main.py
- **Simplified profiling.** Profile now directly instantiates SingleImageCropGenerator instead of going through dynamic importlib dispatch; checkpoint loading always renames the Dense head and loads backbone weights by name, removing a fragile try/except
- **Bolstered documentation.** We added module docstrings and method docstrings to profiling.py, cropping.py, and all retained dataset/ and imaging/ modules
- **Updated README.** We added sections covering how inference works, image preparation (illumination correction + compression), large-scale parallel profiling with split, and verifying the installation
- **Updated ROADMAP and AGENTS.md.** We aligned version numbers (v0.5.1 current, v0.6.x PyTorch rewrite), described the agentic AI vision, and documented scope expectations for contributors
Reformatted all markdown files to one sentence per line
- **Dependency fixes.** Very narrow, fragile range but it works. We pinned numpy<2.0 to fix TF 2.14 / ml-dtypes ABI incompatibility; added tensorflow-macos as an explicit platform-conditional dependency so uv sync --frozen works on macOS arm64 CI runners
- **Integration test suite.** We added @pytest.mark.integration tests that download the Cell Painting CNN v1 checkpoint from Zenodo, verify the 1280-d pool5 embedding, and run the full Profile pipeline end-to-end; added a dedicated zenodo-integration-test CI job that runs these separately from the main matrix

### CODE REVIEW

**Comment by gwaybio** [comment]
@d33bs - please ignore for now

**Comment by gwaybio** [comment]
@d33bs - this is now ready for your eyes. This was a much heavier lift than I naively expected (see my first failed attempt #367 ) Essentially, for any DeepProfiler PyPI release to be valid, we needed to overhaul the project sooner than I had estimated. See above for updated PR description

**Comment by gwaybio** [comment]
@jccaicedo - tagging you for visibility. If you're able, we'd benefit from your comments as well. Thanks!

File: .github/workflows/integration-test.yml [hunk 1]
```diff
@@ -43,7 +43,7 @@ jobs:
         run: uv run ruff check .
 
   python-type-checks:
-    name: Python type checks (ty)
+    name: Python type checks (mypy)
```
Comment (d33bs) [inline .github/workflows/integration-test.yml]: Looks like we slid back to mypy. I don't suggest this as a sustainable path forward, there are just too many edge cases which mypy forces.

File: .github/workflows/integration-test.yml [hunk 2]
```diff
@@ -116,6 +115,28 @@ jobs:
         if: (matrix.os != 'ubuntu-24.04') || (matrix.python-version != env.TARGET_PYTHON_VERSION)
         run: uv run pytest ${{ github.event.inputs.pytest_addopts }}
 
+  zenodo-integration-test:
```
Comment (d33bs) [inline .github/workflows/integration-test.yml]: Keeping this clear (Zenodo archive integration issues are top of mind for me and initially misread).
```suggestion
  zenodo-model-integration-test:
```

File: README.md [hunk 3]
```diff
@@ -46,9 +42,10 @@ robust and improve performance.
 
 ## System requirements
 
-- Python 3.10+ is required.
-- Linux (Ubuntu 20.04+) or macOS.
-- For GPU acceleration, a CUDA-compatible GPU is recommended.
+- Python 3.10 or 3.11
```
Comment (d33bs) [inline README.md:45]: Later Python version compatibility should be a priority for us on this project soon, Python 3.10 will be EoL this year, meaning we only have one version of compatibility afterwards.

File: deepprofiler/dataset/illumination_statistics.py [hunk 4]
```diff
@@ -11,21 +28,54 @@
...
 
-#################################################
-## COMPUTATION OF ILLUMINATION STATISTICS
-#################################################
-
-# Build pixel histogram for each channel
 class IlluminationStatistics():
+    """Accumulate pixel statistics across all images in one plate.
+
+    Builds a per-channel pixel histogram and a mean image (at downscaled
```
Comment (d33bs) [inline deepprofiler/dataset/illumination_statistics.py:61]: This sounds useful!

File: deepprofiler/dataset/image_dataset.py [hunk 5]
```diff
@@ -198,10 +302,28 @@ def number_of_records(self, dataset):
...
 
     def add_target(self, new_target):
+        """Append a :class:`~deepprofiler.dataset.target.MetadataColumnTarget` to ``self.targets``."""
         self.targets.append(new_target)
 
-def read_dataset(config, mode = 'train'):
-    # Read metadata and split dataset in training and validation
+
+def read_dataset(config, mode='train'):
+    """Build an :class:`ImageDataset` from a config dict.
```
Comment (d33bs) [inline deepprofiler/dataset/image_dataset.py:310]: Seeing the missing docstrings replaced: consider adding the [pydocstring ruff linter config](https://docs.astral.sh/ruff/rules/undocumented-public-function/) to the pyproject.toml.

File: AGENTS.md [hunk 6]
```diff
@@ -25,7 +26,7 @@ Always work within the uv-managed environment.
 
 - **Do not modify deprecated code** unless fixing a bug that affects the current release.
   Deprecated paths are: `deepprofiler/learning/`, `deepprofiler/imaging/augmentations.py`, `deepprofiler/imaging/cropping.py`, `deepprofiler/dataset/sampling.py`, `plugins/`.
-- **Do not anticipate the v0.4.x rewrite** in v0.3.x changes.
+- **Do not anticipate the v0.6.x PyTorch rewrite** in v0.5.1 changes.
```
Comment (d33bs) [inline AGENTS.md]: I feel like we should avoid version specifics in the agents file. Perhaps this is better suited for the roadmap with strong guidance here to abide what's in the roadmap as per existing git tags and upstream repo status?

File: deepprofiler/profiling.py [hunk 7]
```diff
@@ -0,0 +1,201 @@
...
+    weights.  The final layers are GlobalAveragePooling2D (named ``pool5``) and
+    a softmax Dense head (named ``ClassProb``).  Weights are loaded separately
+    via :meth:`Profile.configure`.
+
+    If ``config["profile"]["use_pretrained_input_size"]`` is set, builds a
+    3-channel ImageNet-pretrained model at that spatial resolution instead —
+    useful for fine-tuning experiments.
+    """
+    num = config["train"]["model"]["params"]["conv_blocks"]
+    assert num in _EFFICIENTNET_MODELS, f"{num} not in supported EfficientNet variants: {list(_EFFICIENTNET_MODELS)}"
```
Comment (d33bs) [inline deepprofiler/profiling.py]: Consider removing these asserts and using if-cases with specific Exceptions instead.

File: tests/deepprofiler/dataset/test_pixels.py [hunk 8]
```diff
@@ -10,23 +10,18 @@
...
-    # Stores the paths to the random images and outlines files in a list and a string
-    test_paths = [tmp_path + "rand_img_1.jpg",tmp_path + "rand_img_2.jpg",tmp_path + "rand_img_3.jpg"]
-    test_outlines = tmp_path + "rand_outlines.jpg"
-    
-    # Creates an image from the random paths and tests the output dimensions
+
+    skimage.io.imsave(tmp_path + "/rand_img_1.png", numpy.random.randint(256, size=(16, 16), dtype=numpy.uint16))
+    skimage.io.imsave(tmp_path + "/rand_img_2.png", numpy.random.randint(256, size=(16, 16), dtype=numpy.uint16))
+    skimage.io.imsave(tmp_path + "/rand_img_3.png", numpy.random.randint(256, size=(16, 16), dtype=numpy.uint16))
+    skimage.io.imsave(tmp_path + "/rand_outlines.png", numpy.random.randint(256, size=(16, 16), dtype=numpy.uint16))
```
Comment (d33bs) [inline tests/deepprofiler/dataset/test_pixels.py:18]: Is there a way to test a real dataset here too? Perhaps something from the publication? It need not be a "full dataset" but having something "real" would probably give us better observation of the code changes. Maybe this is something to think about including in OME-IRIS too, eventually.

File: deepprofiler/profiling.py (hunk 7 above)
Comment (d33bs) [reply deepprofiler/profiling.py]: Also, we can include another ruff linter rule to avoid these: https://docs.astral.sh/ruff/rules/assert

**Review by d33bs — APPROVED** [review]
Nice! Looking good to me.

File: .github/workflows/integration-test.yml (hunk 1 above)
Comment (gwaybio) [reply .github/workflows/integration-test.yml]: i was unimpressed by `ty`! It was providing many false positives, which `mypy` avoided. I'll drop it now though, heeding your warnings.

File: README.md (hunk 3 above)
Comment (gwaybio) [reply README.md:45]: yes, agree... if we were a bit later in revitalizing this project, we would have had to use a different strategy given the narrowing window of dependency support

File: deepprofiler/dataset/illumination_statistics.py (hunk 4 above)
Comment (gwaybio) [reply deepprofiler/dataset/illumination_statistics.py:61]: yes! I meant to tag @jenna-tomkinson 👀

File: deepprofiler/dataset/image_dataset.py (hunk 5 above)
Comment (gwaybio) [reply deepprofiler/dataset/image_dataset.py:310]: thanks, added!

File: AGENTS.md (hunk 6 above)
Comment (gwaybio) [reply AGENTS.md]: yep, i agree - i'll update this file

File: deepprofiler/profiling.py (hunk 7 above)
Comment (gwaybio) [reply deepprofiler/profiling.py]: added, thanks!!

File: tests/deepprofiler/dataset/test_pixels.py (hunk 8 above)
Comment (gwaybio) [reply tests/deepprofiler/dataset/test_pixels.py:18]: added in ae4db39b7f04b0074de85422850a413b312bfdf5 - thanks for this suggestion!

**Comment by gwaybio** [comment]
thanks for the review @d33bs ! I'll keep this PR open for ~1 week to give @jccaicedo a chance to comment. Juan, please let us know if you need more time!

File: deepprofiler/dataset/illumination_statistics.py (hunk 4 above)
Comment (jenna-tomkinson) [reply deepprofiler/dataset/illumination_statistics.py:61]: WHOA! I didn't realize DeepProfiler offered illumination correction. I think this is something we can test to see the performance compared to CellProfiler, especially since this looks to be fairly parameter-free. I wonder why this is kept under DeepProfiler though, feels like this could be an independent software for IC in cytomining, but just my first impression.

**Comment by jccaicedo** [comment]
This looks fantastic! Thanks for refactoring it in this way. It will be much more useful with these changes!

File: deepprofiler/dataset/illumination_statistics.py (hunk 4 above)
Comment (d33bs) [reply deepprofiler/dataset/illumination_statistics.py:61]: @jenna-tomkinson, @gwaybio - should we start another project to capture this value for corrective image transformation? I can take lead here with a naïve approach to start, we can refine it over time.

Comment (jenna-tomkinson) [reply deepprofiler/dataset/illumination_statistics.py:61]: Personally, I do not see this needing to live in DeepProfiler, since from my understanding you were meant to process images and get centroids prior. 

I agree that it should be its own thing so that someone can call it and use it outside of CellProfiler (might be faster?). But, not sure how long it would take to implement.

**Comment by gwaybio** [comment]
thank you @jccaicedo ! I'll merge this in now, which will trigger pypi release
